### ISSUE TITLE
Check for unused injected values

### ISSUE THREAD

**Issue by remcohaszing** [Gillespie59/eslint-plugin-angular#159]
If injected values are unused, this is mostly a result of forgetting to remove them.

The following should trigger a warning:
```js
factory('example', function(dependency) {})
```

The following should not trigger a warning:
```js
factory('example', function(dependency) {
  return dependency;
})

factory('example', function(dependency) {
  dependency();
})

factory('example', function(dependency) {
  function nested() {dependency();}
})
```

The rule should apply to all places where dependencies may be injected:

- factories
- services
- controllers
- directives
- decorators
- run
- config
- providers
- providers `$get` function.
- functions annotated with `/*@ngInject*/`

**Comment by alexlrobertson** [comment]
Any reason [`no-unused-vars`](http://eslint.org/docs/rules/no-unused-vars) doesn't cover this for you?

**Comment by remcohaszing** [comment]
`no-unused-vars` checks for unused variables. Typically functions may accept unused variables to pad the actually used variables. When injecting using AngularJS the variables are not used for padding, they are just unused.

`no-unused-vars` is triggered for this example:
```js
factory('example', function($q, $http) {
  return $q.resolve();
})
```

But not for this example:
```js
factory('example', function($http, $q) {
  return $q.resolve();
})
```

**Comment by alexlrobertson** [comment]
You can set the "args" option to "all" and it will check for use across all parameters. This however presents issues with patterns where the parameters are fixed (i.e. directives).

```js
function (scope, element, attrs) {
    scope.foo = attrs.bar;
}
```

I agree that this is an issue I'd like solved, but I think the cases where it applies are difficult to suss out.
